File: app/src/main/java/com/books/quickseries/ui/base/BaseActivity.java
package com.books.quickseries.ui.base;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

/**
 * Created by zakir on 06/09/2018.
 */

public abstract class BaseActivity extends AppCompatActivity {

    public abstract void initUI();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onResume() {
        super.onResume();
        initUI();
    }
}

File: app/src/main/java/com/kotlin/test/ui/BaseActivity.kt
package com.kotlin.test.ui

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.app.AppCompatDelegate
import com.kotlin.test.application.TestApplication
import com.kotlin.test.di.component.ActivityComponent
import com.kotlin.test.di.module.ActivityModule

open class BaseActivity : AppCompatActivity(){
    var activityComponent: ActivityComponent? = null
        private set

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true)
        activityComponent = DaggerActivityComponent.builder()
            .activityModule(ActivityModule(this))
            .applicationComponent((application as TestApplication).component)
            .build()
    }
}